src/js/components/DateInput/DateInput.js

```diff
@@ -68,22 +68,22 @@ const DateInput = forwardRef(
       initialValue: defaultValue,
     });
 
-    let timestamp;
-    if (Array.isArray(defaultValue) && defaultValue.length) {
-      timestamp = getTimestamp(defaultValue[0]);
-    } else if (typeof defaultValue === 'string') {
-      timestamp = getTimestamp(defaultValue);
-    } else if (Array.isArray(value) && value.length) {
-      timestamp = getTimestamp(value[0]);
+    const timestamp = useMemo(() => {
+      if (Array.isArray(defaultValue) && defaultValue.length)
+        return getTimestamp(defaultValue[0]);
+      if (typeof defaultValue === 'string') return getTimestamp(defaultValue);
+      if (Array.isArray(value) && value.length) return getTimestamp(value[0]);
       // check to see if value is not an empty string
       // empty string should behave like undefined
-    } else if (typeof value === 'string' && value.length) {
-      timestamp = getTimestamp(value);
-    }
+      if (typeof value === 'string' && value.length) return getTimestamp(value);
+      return undefined;
+    }, [defaultValue, value]);
 
-    // normalize value based on timestamp vs user's local timezone
-    const normalizedDate = normalizeForTimezone(value, timestamp);
+    // whether or not we should normalize the date based on the timestamp
```

Comment:
maybe beef up this comment just a bit to explain when it is set false?

Reply:
Done!